gramlib/grammar.ml

```diff
@@ -153,6 +156,7 @@ and ('trecs, 'trecp, 'a) ty_rec_level = {
 
 and ('self, 'trec, 'a) ty_symbol =
 | Stoken : 'c pattern -> ('self, norec, 'c) ty_symbol
+| Stokens : ty_pattern list -> ('self, norec, unit) ty_symbol
```

Comment:
It looks a bit dangerous to me to have a redundancy between `Stoken` and `Stokens`. Maybe we should merge both nodes in a way that ensures that `Stokens` of a singleton has the same typing rule as before.

Reply:
The redundancy is not nice indeed, but they have different types what justifies to have both. I agree that it is a compromise.

Reply:
You could type Stokens with a product type of the subrules so that when they have a length of one, they have the same rules a before. I don't think it is necessary for this PR though, it can always be refined later.